File: spritebench_cs/Sprite.cs
using Godot;
using System;

public partial class Sprite : Sprite2D
{
	private static readonly Texture2D Icon = GD.Load<Texture2D>("res://icon.svg");
	private static readonly Vector2 Size2 = Icon.GetSize() / 2;

	private float _angle = (float)GD.RandRange(0, Mathf.Tau);
	private float _speed = (float)GD.RandRange(100, 600);
	private Vector2 _pos;
	private Vector2 _windowSize;

	public override void _Ready()
	{
		Texture = Icon;
		Position = (Vector2)GetWindow().Size / 2;
		_pos = Position;
		_windowSize = (Vector2)GetWindow().Size;
	}

	public override void _Process(double delta)
	{
		_pos += new Vector2(Mathf.Cos(_angle), Mathf.Sin(_angle)) * _speed * (float)delta;
		Position = _pos;

		if (Position.X < Size2.X || Position.X > _windowSize.X - Size2.X)
		{
			_angle = Mathf.Pi - _angle;
		}
		if (Position.Y < Size2.Y || Position.Y > _windowSize.Y - Size2.Y)
		{
			_angle = -_angle;
		}
	}
}

File: spritebench_swift/swift_godot_game/Sources/SpriteBenchSwift/Sprite.swift
import Foundation
import SwiftGodot

@Godot
class Sprite: Sprite2D {
    var angle: Float = .random(in: 0...(Float.pi * 2))
    var speed: Float = .random(in: 100...600)
    var pos: Vector2 = .zero
    var windowSize: Vector2 = .zero
    var halfSize: Vector2 = .zero

    override func _process(delta: Double) {
        pos = pos + Vector2(x: cos(angle), y: sin(angle)) * Double(speed) * delta
        position = pos

        if pos.x < halfSize.x || pos.x > windowSize.x - halfSize.x {
            angle = Float.pi - angle
        }
        if pos.y < halfSize.y || pos.y > windowSize.y - halfSize.y {
            angle = -angle
        }
    }
}
